llvm/test/CAS/thinlto-remote-cache.ll

```diff
@@ -0,0 +1,42 @@
+; REQUIRES: remote-cache-service
+
+; RUN: opt -module-hash -module-summary %s -o %t.bc
+; RUN: opt -module-hash -module-summary %p/Inputs/cache.ll -o %t2.bc
+
+; RUN: rm -f %{remote-cache-dir}/%basename_t
+; RUN: rm -rf %t && mkdir -p %t
+
+; RUN: llvm-remote-cache-test -socket-path=%{remote-cache-dir}/%basename_t -cache-path=%t/cache -- \
+; RUN:    llvm-lto -thinlto-action=run -exported-symbol=globalfunc %t2.bc %t.bc -thinlto-cache-dir grpc:%{remote-cache-dir}/%basename_t
+
+; RUN: llvm-cas --cas %t/cache/cas --dump | FileCheck %s
+
+;; Check the cas dump, expect 4 items.
+; CHECK: index:
+; CHECK: root
+; CHECK: records
+; CHECK-NEXT: OK=object  SK=datapool
+; CHECK-NEXT: OK=object  SK=datapool
+; CHECK-NEXT: OK=object  SK=datapool
+; CHECK-NEXT: OK=object  SK=datapool
+; CHECK-EMPTY:
+; CHECK-NEXT: pool:
+
+; RUN: llvm-remote-cache-test -socket-path=%{remote-cache-dir}/%basename_t -cache-path=%t/cache -- \
+; RUN:    llvm-lto -thinlto-action=run -exported-symbol=globalfunc %t2.bc %t.bc -thinlto-cache-dir grpc:%{remote-cache-dir}/%basename_t
+
+;; CAS should remain the same.
+; RUN: llvm-cas --cas %t/cache/cas --dump | FileCheck %s
+
+;; Check save object file path also works
+; RUN: llvm-remote-cache-test -socket-path=%{remote-cache-dir}/%basename_t -cache-path=%t/cache -- \
+; RUN:    llvm-lto -thinlto-action=run -exported-symbol=globalfunc %t2.bc %t.bc -thinlto-cache-dir grpc:%{remote-cache-dir}/%basename_t -thinlto-save-objects %t/objects
+; RUN: llvm-cas --cas %t/cache/cas --dump | FileCheck %s
+
```

Comment:
Should we also have a test invocation that uses the environment variable for setting up the remote cache?

Reply:
Not with the current implementation as the environment variable overwrite is only in libLTO, so it requires `ld64` to test. I am not too worried about that.